frontend/src/metabase/search/types.ts

```diff
@@ -29,12 +31,17 @@ export type FilterTypeKeys = keyof SearchFilterPropTypes;
 
 export type SearchFilters = Partial<SearchFilterPropTypes>;
 
-export type SearchFilterComponent<T extends FilterTypeKeys = any> = FC<
-  {
-    value?: SearchFilterPropTypes[T];
-    onChange: (value: SearchFilterPropTypes[T]) => void;
-    "data-testid"?: string;
-  } & Record<string, unknown>
->;
+export type SearchFilterComponentProps<T extends FilterTypeKeys = any> = {
+  value?: SearchFilterPropTypes[T];
+  onChange: (value: SearchFilterPropTypes[T]) => void;
+  "data-testid"?: string;
+} & Record<string, unknown>;
 
 export type SearchAwareLocation = Location<{ q?: string } & SearchFilters>;
+
+export type SearchSidebarFilterComponent<T extends FilterTypeKeys = any> = {
```

Comment:
Modified the type here so that we can have a common interface for all of the filters (except `Verified`, which will be an exception because it's a toggle rather than a dropdown). The `SidebarFilter` component will take these props and use them to display and transact data uniformly, and all we need to worry about is the content of the dropdown and the display once an element has been selected